analyzers/src/SonarAnalyzer.Common/Rules/HardcodedIpAddressBase.cs

```diff
@@ -20,74 +20,130 @@
 
 using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Linq;
 using System.Net;
 using System.Net.Sockets;
+using System.Text.RegularExpressions;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.Diagnostics;
 using SonarAnalyzer.Common;
 using SonarAnalyzer.Helpers;
 
 namespace SonarAnalyzer.Rules
 {
-    public abstract class HardcodedIpAddressBase<TLiteralExpression> : HotspotDiagnosticAnalyzer
+    public abstract class HardcodedIpAddressBase<TSyntaxKind, TLiteralExpression> : HotspotDiagnosticAnalyzer
+        where TSyntaxKind : struct
         where TLiteralExpression : SyntaxNode
     {
+        private static bool IsBroadcast(string ip) =>
+            ip.Equals("255.255.255.255", StringComparison.InvariantCultureIgnoreCase);
+
+        private static bool DoesItLookLikeObjectIdentifier(string s) =>
+            s.StartsWith("2.5.");
+
         protected const string DiagnosticId = "S1313";
+
         protected const string MessageFormat = "Make sure using this hardcoded IP address '{0}' is safe here.";
 
-        private static readonly ISet<string> IgnoredVariableNames =
+        private const int NumOfIPv4AddressParts = 4;
```

Comment:
Just to simplify the name. It's integer, so it's obvious that it's a count
```suggestion
        private const int IPv4AddressParts = 4;
```